### PR TITLE
feat(sessions): Enable session tracking by default

### PR DESCRIPTION
**Pull request by bengourley**
## Goal

Enable session tracking by default.

## Changeset

To facilitate session tracking defaulting to on, this changeset also updates the way in which the notify/sessions endpoint URLs are configured. "useSSL", "notifyHost", "notifyPath", "notifyPort" are deprecated, as is "sessionEndpoint". 

The new way to configure endpoint URLs is in the "endpoints" option. Using the deprecated options will still work but a message will be output to the console.

## Tests

New tests were added for the different permutations of configuring endpoint URLs with current and deprecated options, and how they interact with the `autoCaptureSessions` option.

## Review

For the submitter, initial self-review:

- [x] Commented on code changes inline explain the reasoning behind the approach
- [x] Reviewed the test cases added for completeness and possible points for discussion
- [x] A changelog entry was added for the goal of this pull request
- [x] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [x] Initial review of the intended approach, not yet feature complete
  - [x] Structural review of the classes, functions, and properties modified
  - [x] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [x] Consistency across platforms for structures or concepts added or modified
- [x] Consistency between the changeset and the goal stated above
- [x] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [x] Usage friction - is the proposed change in usage cumbersome or complicated?
- [x] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [x] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [x] Thoroughness of added tests and any missing edge cases
- [x] Idiomatic use of the language

### CODE REVIEW

File: lib/sessions/tracker.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ module.exports = class SessionTracker extends Emitter {
   }
 
   start () {
-    if (!this._interval) this._interval = setInterval(this._summarize, this._intervalLength)
+    if (!this._interval) this._interval = setInterval(this._summarize, this._intervalLength).unref()
```
Comment (bengourley) [inline lib/sessions/tracker.js]: when not calling `.unref()` on this interval, bugsnag has the side effect of keeping the process running. This is because it always has something on the event loop. By calling `unref()` it means if all other timeouts/intervals/callbacks complete, the process can terminate as it usually would.

Comment (kattrali) [reply lib/sessions/tracker.js]: This possibly warrants its own change log entry
